internal/agentdeployer/agent.go

```diff
@@ -261,18 +261,39 @@ func (d *DockerComposeAgentDeployer) installDockerCompose(agentInfo AgentInfo) (
 			return "", fmt.Errorf("failed to obtain has for Elastic Agent Dockerfile: %w", err)
 		}
 	}
+	config, err := stack.LoadConfig(d.profile)
+	if err != nil {
+		return "", fmt.Errorf("failed to load config from profile: %w", err)
+	}
+
+	fleetURL := "https://fleet-server:8220"
+	kibanaHost := "https://kibana:5601"
+	stackVersion := d.stackVersion
+	if config.Provider == stack.ProviderServerless {
+		fleetURL = config.Parameters[stack.ParamServerlessFleetURL]
+		kibanaHost = config.KibanaHost
+		stackVersion = config.Parameters[stack.ParamServerlessLocalStackVersion]
+	}
```

Comment:
Nit. Couldn't we get this info from the config in all cases?
```suggestion
	fleetURL = config.Parameters[stack.ParamServerlessFleetURL]
	kibanaHost = config.KibanaHost
	stackVersion = config.Parameters[stack.ParamServerlessLocalStackVersion]

```

Reply:
Currently, it cannot be done. When compose provider is used, kibanaHost is `https://127.0.0.1:5601`, so it cannot be used to set the environment variable for the Elastic Agent. It requires the container name `kibana`.

In case of fleet-server, that value is not available as a configuration parameter now in the Compose provider. Probably, it could be added to the compose provider, but it we do so, it should be renamed (`stack_fleet_url`?) or create a new parameter for the compose provider, eg. `compose_fleet_url`.

Should I try to use just one parameter for both providers? @jsoriano 

This is the current definition of this parameter for serverless: https://github.com/elastic/elastic-package/blob/c2a0f6df3eeffee964c8c2b1ee4c5e96d2df464c/internal/stack/serverless.go#L27

Reply:
Well, lets leave it like this by now, maybe we can implement this logic in the provider later.